hang in shm/test_ucp_fence64.atomic_add_fadd/2

http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4826/label=hpc-test-node,worker=1/console
```
02:07:12 
02:07:12 [----------] 1 test from shm/test_ucp_fence64
02:07:12 [ RUN      ] shm/test_ucp_fence64.atomic_add_fadd/2
04:27:53 Build timed out (after 150 minutes). Marking the build as failed.
04:27:53 Build was aborted
04:27:53 TAP Reports Processing: START
04:27:53 Looking for TAP results report in workspace using pattern: **/*.tap
04:27:53 Saving reports...
04:27:53 Processing '/scrap/jenkins/jobs/hpc-ucx-pr/configurations/axis-label/hpc-test-node/axis-worker/1/builds/4826/tap-master-files/build-test/mpi_tests.tap'
04:27:53 Parsing TAP test result [/scrap/jenkins/jobs/hpc-ucx-pr/configurations/axis-label/hpc-test-node/axis-worker/1/builds/4826/tap-master-files/build-test/mpi_tests.tap].
04:27:53 TAP Reports Processing: FINISH
04:27:53 Archiving artifacts
04:27:53 
make: *** [test] Terminated
```

--- a/test/gtest/ucp/test_ucp_fence.cc
+++ b/test/gtest/ucp/test_ucp_fence.cc
@@ -142,9 +142,6 @@ protected:
         uint32_t error = 0;
 
         sender().connect(&receiver(), get_ep_params());
-        if (&sender() != &receiver()) {
-            receiver().connect(&sender(), get_ep_params());
-        }
 
         params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
                             UCP_MEM_MAP_PARAM_FIELD_LENGTH |

